### PR TITLE
feat(config): Views

### PR DESCRIPTION
**Pull request by ksqsf**
image

Click `Print` to see updated values in stdout.

Some known issues:
- Lists are invisible in ScrollView
- Element order unstable
- Key not supported (will need https://github.com/Kentzo/ShortcutRecorder)

They will be addressed in later PRs.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,16 @@
-file(GLOB CONFIG_UI_FILES config/*.swift)
-
 add_executable(Fcitx5
     MACOSX_BUNDLE
     server.swift
     controller.swift
-    ${CONFIG_UI_FILES}
+    config/config.cpp
```
Comment (eagleoflqj) [inline src/CMakeLists.txt:5]: If all swift files are listed I think using glob is better.

File: .gitmodules [hunk 2]
```diff
@@ -4,3 +4,6 @@
 [submodule "fcitx5-webview"]
 	path = fcitx5-webview
 	url = https://github.com/fcitx-contrib/fcitx5-webview
+[submodule "deps/SwiftyJSON"]
+	path = deps/SwiftyJSON
+	url = https://github.com/SwiftyJSON/SwiftyJSON.git
```
Comment (eagleoflqj) [inline .gitmodules:9]: The trailing .git is not needed.

File: src/config/config.swift [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  static func decode(json: JSON) throws -> Self {
+    if let decoded = json.string {
+      return decoded
+    } else {
+      throw FcitxCodingError.invalidArgument(context: json)
+    }
+  }
+  func encode() -> String {
+    let encodedString = self.replacingOccurrences(of: "\"", with: "\\\"")
+      .replacingOccurrences(of: "\\", with: "\\\\")
```
Comment (eagleoflqj) [inline src/config/config.swift:154]: This will change `"` to `\\"`, is that expected?

Comment (ksqsf) [reply src/config/config.swift:154]: No. Good catch!

**Review by eagleoflqj — APPROVED** [review]
Nice start!

**Pull request merged**